### DISCUSSION TITLE
wp:pattern block inside a block pattern with the same slug are linked when the block pattern is inserted

### DISCUSSION THREAD

**Discussion by ChrissiQ** [WordPress/gutenberg#50554] [Q&A] [answered]
I don't know if this is intended or not but I'm confused by it.

I created a block pattern named `mynamespace/glossary-group`.  Then I created a new block pattern named `mynamespace/glossary-list` which includes 4 instances of `<!-- wp:pattern {"slug":"mynamespace/glossary-group"} /-->`.  When I insert the `mynamespace/glossary-list`, the 4 instances of `mynamespace/glossary-group` are linked. When I select one, all of them are selected.  When I edit the content inside one, the content inside the others is linked.

Then, once the page is saved and editor reloaded, they are no longer linked.

Is this intentional?  Can I make the multiple instances of the same pattern inside a block pattern unlinked somehow?

**Comment by talldan** [comment]
@ChrissiQ Thanks for mentioning this. It looks like a bug.

When multiple pattern blocks use the same slugs, the blocks that are created all use the same block ids (clientIds), so this weird behavior happens.

Kudos to @aaronrobertshaw for talking it through with me and spotting the issue!

**Comment by talldan** [comment, marked as answer]
I've created a bug report here - #50628. 

And a fix here - #50629

(edit: now merged)
